pkg/syncer/admissionhandler/admissionhandler.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admissionhandler
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+)
+
+type (
+	parameterSet map[string]struct{}
+)
+
+// Has checks if specified paramName present in the paramSet
+func (paramSet parameterSet) Has(paramName string) bool {
+	_, ok := paramSet[paramName]
+	return ok
+}
+
+var (
+	webhookServerPort = 8443
```

Comment:
Can we change the default port to something else (say 9443), as 8443 is already being used for Authproxy in 7.0u1 ?

Reply:
Can we make these ports configurable?

Reply:
It is configurable. If the admission-webhook-port value is not supplied in config, then it picks up this default value.

Reply:
@gohilankit we have not used port `8443` in our pod anywhere. If `8443` is used in the other pod I think we are good.
As we bind our service to our controller pod only. And also this port can be configured using vSphere config secret.